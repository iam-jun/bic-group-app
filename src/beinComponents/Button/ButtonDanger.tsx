import React from 'react';
import {
  StyleProp,
  StyleSheet,
  TouchableHighlight,
  ViewStyle,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import ButtonWrapper, {
  ButtonWrapperProps,
} from '~/beinComponents/Button/ButtonWrapper';

import spacing from '~/theme/spacing';

export interface ButtonDangerProps extends ButtonWrapperProps {
  useI18n?: boolean;
}

const ButtonDanger: React.FC<ButtonDangerProps> = ({
  useI18n,
  children,
  style,
  disabled,
  ...props
}: ButtonDangerProps) => {
  const { colors }: ExtendedTheme = useTheme() as ExtendedTheme;

  const underlayColor = colors.neutral5;
  let backgroundColor = colors.red1;
  let textColor = colors.red60;

  if (disabled) {
    backgroundColor = colors.gray20;
    textColor = colors.gray40;
  }

  const containerStyle: StyleProp<ViewStyle> = StyleSheet.flatten([
    {
      backgroundColor,
      padding: spacing.padding.small,
      borderRadius: spacing.borderRadius.small,
      alignItems: 'center',
    },
    style,
  ]);

  return (
    <ButtonWrapper
      disabled={disabled}
      style={containerStyle}
      underlayColor={underlayColor}
      TouchableComponent={TouchableHighlight}
      {...props}
      textProps={{ color: textColor, useI18n, ...props?.textProps }}
    >
      {children}
    </ButtonWrapper>
  );
};

export default ButtonDanger;
