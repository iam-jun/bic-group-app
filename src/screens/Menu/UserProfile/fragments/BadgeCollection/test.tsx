import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
