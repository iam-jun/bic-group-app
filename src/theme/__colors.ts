// Color system
export const white = '#ffffff';
export const grey1 = '#f4f4f4';
export const grey2 = '#EBEAEC';
export const grey3 = '#dee2e6';
export const grey4 = '#adb5bd';
export const grey5 = '#999999';
export const grey6 = '#777777';
export const grey7 = '#383838';
export const grey8 = '#1e1e1e';
export const grey9 = '#2C2C2C';
export const black = '#000';

export const red = '#ff5148';
export const orange = '#F2711C';
export const yellow = '#FBBD08';
export const olive = '#B5CC18';
export const green = '#4AD498';
export const teal = '#00B5AD';
export const blue = '#2185D0';
export const violet = '#6435C9';
export const purple = '#A333C8';
export const pink = '#E03997';
export const brown = '#A5673F';
export const accent = '#6600cc';
export const primary = '#220b3b';

export default {
  dark: {
    colors: {
      accent: '#6600cc',
      native: '#2A2C36',
      backdrop: 'rgba(0, 0, 0, 0.5)',
      background: grey7,
      disabled: 'rgba(255, 255, 255, 0.38)',
      error: '#E84D4B',
      lightRed: '#E84D4B',
      notification: '#ff80ab',
      onBackground: '#FFFFFF',
      onSurface: '#FFFFFF',
      placeholder: 'rgba(255, 255, 255, 0.54)',
      primary: '#fff',
      surface: black,
      text: '#FFFFFF',
      bgColor: grey8,
      bgColorSecondary: '#220b3b',
      warning: yellow,
      innerButton: '#FFFFFF',
      buttonBackground: '#220b3b',
      hightLight: '#59bbc4',
      textSecondary: '#7fe5f0',
      tag: grey7,
      red,
      orange,
      yellow,
      olive,
      green,
      blue,
      teal,
      violet,
      purple,
      pink,
      brown,
      grey1,
      grey2,
      grey3,
      grey4,
      grey5,
      grey6,
      grey7,
      grey8,
      grey9,
      black,
      //TabBar
      tabBarBackground: black,
      activeColor: '#F7F7F7',
      inactiveColor: grey6,
      // ViewLabel
      ViewLabel: {
        color: grey6,
        colorHeading: white,
      },
      secondary: grey6,
      white,
      greyOutline: '#bbb',
      searchBg: '#303337',
      listItemBg: grey8,
      success: green,
      divider: grey7,
      border: grey9,
      icon: grey7,
      avatar: '#b28a5c',
      platform: {
        ios: {
          primary: '#007aff',
          secondary: '#5856d6',
          success: '#4cd964',
          error: '#ff3b30',
          warning: '#ffcc00',
        },
        android: {
          primary: '#2196f3',
          secondary: '#9C27B0',
          success: '#4caf50',
          error: '#f44336',
          warning: '#ffeb3b',
        },
      },
      headerStyle: {
        backgroundColor: '#1e1f28',
        headerTintColor: '#ffffff',
      },
      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',
    },
  },
  light: {
    colors: {
      accent: '#6600cc',
      native: '#E5E5E5',
      backdrop: 'rgba(0, 0, 0, 0.5)',
      background: '#f6f6f6',
      disabled: 'rgba(0, 0, 0, 0.26)',
      error: '#E84D4B',
      lightRed: '#E84D4B',
      notification: '#f50057',
      onBackground: '#252D42',
      onSurface: '#000000',
      placeholder: 'rgba(0, 0, 0, 0.54)',
      primary: '#220b3b',
      surface: '#ffffff',
      text: '#252D42',
      bgColor: white,
      bgColorSecondary: '#f6f6f6',
      warning: yellow,
      icon: grey3,
      avatar: '#707f9f',
      innerButton: '#368C8B',
      buttonBackground: '#EBEAEC',
      hightLight: '#59bbc4',
      tag: grey3,
      textSecondary: '#00b9cf',
      red,
      orange,
      yellow,
      olive,
      green,
      blue,
      teal,
      violet,
      purple,
      pink,
      brown,
      grey9,
      //TabBar
      tabBarBackground: '#fff',
      activeColor: '#220b3b',
      inactiveColor: '#adb5bd',
      // ViewLabel
      ViewLabel: {
        color: grey5,
        colorHeading: black,
      },
      secondary: grey6,
      white: white,
      grey1: grey1,
      grey2: grey2,
      grey3: grey3,
      grey4: grey4,
      grey5: grey5,
      grey6: grey6,
      grey7: grey7,
      grey8: grey8,
      black: black,

      greyOutline: '#bbb',
      searchBg: '#303337',
      listItemBg: white,
      success: green,
      // Darker color if hairlineWidth is not thin enough
      divider: grey2,
      border: grey2,
      platform: {
        ios: {
          primary: '#007aff',
          secondary: '#5856d6',
          success: '#4cd964',
          error: '#ff3b30',
          warning: '#ffcc00',
        },
        android: {
          primary: '#2196f3',
          secondary: '#9C27B0',
          success: '#4caf50',
          error: '#f44336',
          warning: '#ffeb3b',
        },
      },
      headerStyle: {
        backgroundColor: '#ffffff',
        headerTintColor: '#000000',
      },
      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#000000',
    },
  },
};
