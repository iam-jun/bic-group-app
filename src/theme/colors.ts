export default {
  dark: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A',
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD',
      primary7: '#421187',

      background: '#20272F',
      bgSecondary: '#2F3945',
      bgFocus: '#505E71',
      bgHover: '#3F4B5B',
      bgDisable: '#505E71',
      bgButtonPrimary: '#8B5CE9',
      bgButtonSecondary: '#20272E',
      BorderButtonSecondary: '#8B5CE9',
      borderFocus: '#627286',
      borderDisable: '#627286',
      borderCard: '#B2BDCD',
      borderDivider: '#3F4B5B',
      textInput: '#F9FAFB',
      placeholder: '#3F4B5B',
      surface: '#2F3945', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#D5DCE5',
      iconTintLight: '#77879C',
      iconTintReversed: '#2F3945',

      link: '#2396FF',
      activeLink: '#5B22B8',
      success: '#0AC458',
      error: '#E8455F',
      warning: '#FFBC1F',
      statusOrange: '#E98A32',

      textPrimary: '#F9FAFB',
      textSecondary: '#77879C',
      textSuccess: '#FFFFFF',
      textDanger: '#FFFFFF',
      textWarning: '#2F3945',
      textInfo: '#007AFF',
      textDisabled: '#627286',
      textReversed: '#2F3945',
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#2F3945', //Neutral0
      activeColor: '#8B5CE9', //Primary50
      inactiveColor: '#77879C', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
  light: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A', //primary is brand color
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD', //Main Brand
      primary7: '#421187', //Primary90

      background: '#FFFFFF', //Neutral0
      bgSecondary: '#F9FAFB', //Neutral1
      bgFocus: '#D5DCE5', //Neutral10
      bgHover: '#EAEDF2', //Neutral5
      bgDisable: '#D5DCE5', //Neutral10
      bgButtonPrimary: '#7843DD ', //Primary50
      bgButtonSecondary: '#FFFFFF', //Neutral0'
      BorderButtonSecondary: '#7843DD',
      borderFocus: '#B2BDCD', //Neutral20
      borderDisable: '#B2BDCD', //Neutral20
      borderCard: '#B2BDCD', //Neutral20
      borderDivider: '#EAEDF2', //`Neutral5
      textInput: '#2F3945', // Neutral80
      placeholder: '#EAEDF2', //Neutral5
      surface: '#F9FAFB', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#2F3945', //Neutral80
      iconTintLight: '#627286', //Neutral50
      iconTintReversed: '#FFFFFF', //Neutral0

      link: '#007AFF', //Blue50
      activeLink: '#421187', //Primary90
      success: '#00A64D', //Green50
      error: '#DD2C3E', //Red50
      warning: '#FFBC1F', //Yellow40
      statusOrange: '#E98A32', //Orange40

      textPrimary: '#2F3945', //Neutral80
      textSecondary: '#627286', //Neutral50
      textSuccess: '#FFFFFF', //Neutral0
      textDanger: '#FFFFFF', //Neutral0
      textWarning: '#2F3945', //Neutral80
      textInfo: '#007AFF', //Blue50
      textDisabled: '#B2BDCD', //Neutral20
      textReversed: '#FFFFFF', //Neutral0
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#FFFFFF', //Neutral0
      activeColor: '#7843DD', //Primary50
      inactiveColor: '#627286', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
};
