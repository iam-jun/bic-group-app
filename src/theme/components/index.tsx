export {default as ThemeView} from './ThemeView';
export {default as Image} from './Image';
export {default as Header} from './Header';
export {default as Text} from './Text';
export {default as SecondaryButton} from './Button/secondary';
export {default as Container} from './Container';
export {default as ViewSpacing} from './ViewSpacing';
