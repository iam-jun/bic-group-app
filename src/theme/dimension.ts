/* Declare Dimensions */
import { Dimensions, Platform } from 'react-native';
import deviceInfoModule from 'react-native-device-info';

const guidelineBaseWidth = 375;

const DeviceWidth = Dimensions.get('window').width;
const DeviceHeight = Dimensions.get('window').height;

// Cover image ratio -> 21:9
export const scaleCoverHeight = (widthSize = DeviceWidth) => (widthSize / 21) * 9;

export const groupProfileImageCropRatio = {
  // image crop ratio for cover photo: 25:11
  backgroundImgUrl: {
    width: 1080,
    height: 475,
  },
  // image crop ratio for avatar: 1:1 -> default, no need to define
  icon: {},
};

export const userProfileImageCropRatio = {
  // image crop ratio for cover photo: 25:11
  backgroundImgUrl: {
    width: 1080,
    height: 475,
  },
  // image crop ratio for avatar: 1:1 -> default, no need to define
  avatar: {},
};

/* Size config used for Text */
export const sizes = {
  h1: 28,
  h2: 24,
  h3: 20,
  h4: 17,
  h5: 15,
  h6: 13,
  subtitleL: 17,
  subtitleM: 15,
  subtitleS: 13,
  subtitleXS: 11,
  bodyM: 15,
  bodyMMedium: 15,
  bodyS: 13,
  bodySMedium: 13,
  bodyXS: 11,
  bodyXSMedium: 11,
  paragraphL: 17,
  paragraphM: 15,
  paragraphS: 13,
  buttonL: 15,
  buttonM: 15,
  buttonS: 13,
  tabL: 15,
  tabM: 13,
  tabS: 11,
  linkM: 15,
  linkS: 13,
  badgeL: 16,
  badgeM: 12,
  badgeS: 10,
  badgeXS: 9,
  labelL: 17,
  labelM: 15,
  labelS: 13,
  dropdownM: 15,
  dropdownS: 13,
  numberM: 15,
  numberS: 13,
  captionS: 11,
  code: 16,
  error: 16,
  mdH1: 22,
  mdH2: 20,
  mdH3: 18,
  mdH4: 16,
  mdParagraph: 14,
};

/* Line Height used for Text */
export const lineHeights = {
  h1: 48,
  h2: 42,
  h3: 36,
  h4: 32,
  h5: 26,
  h6: 24,
  subtitleL: 26,
  subtitleM: 24,
  subtitleS: 20,
  subtitleXS: 16,
  bodyM: 24,
  bodyMMedium: 24,
  bodyS: 20,
  bodySMedium: 20,
  paragraphL: 28,
  paragraphM: 24,
  paragraphS: 22,
  buttonL: 26,
  buttonM: 24,
  buttonS: 20,
  tabL: 26,
  tabM: 24,
  tabS: 20,
  linkM: 22,
  linkS: 20,
  badgeL: 22,
  badgeM: 20,
  badgeS: 16,
  badgeXS: 14,
  labelL: 26,
  labelM: 22,
  dropdownM: 22,
  dropdownS: 20,
  numberM: 22,
  numberS: 20,
  captionS: 16,
  code: 24,
  error: 24,
};

export const letterSpacing = {
  base: 0,
  h1: 0,
  h2: 0,
  h3: 0,
  h4: 0,
  h5: 0,
  h6: 0,
  bodyM: 0,
  bodyS: 0,
  error: 0,
  code: 0,
};

/* Device Dimensions */
export const deviceDimensions = {
  phone: 375,
  smallTablet: 600,
  bigTablet: 768,
};

/* Used for button size width (short, medium, long, max) */
export const sizeButton = {
  short: {
    width: 113,
  },
  medium: {
    width: 158,
  },
  long: {
    width: 217,
  },
  max: {
    width: '100%',
  },
  height: 44,
};

export const avatarSizes = {
  tiny: 24,
  xSmall: 28,
  small: 32,
  base: 40,
  medium: 48,
  large: 64,
  xLarge: 80,
};

export const avatarBorderWidth = {
  tiny: 1,
  xSmall: 1,
  small: 1,
  base: 1,
  medium: 2,
  large: 2,
  xLarge: 2,
};

export const headerHeight = 60;
export const primaryItemHeight = 64;
export const commentBarHeight = 44;
export const postToolbarHeight = 52;
export const maxNewsfeedWidth = 584;
export const bottomBarHeight = 52;
export const permissionRoleSectionHeaderHeight = 56;

export const homeHeaderLogoHeight = 48;
export const homeHeaderTabHeight = 44;
export const homeHeaderHeight = homeHeaderLogoHeight + homeHeaderTabHeight;
const isPhoneWithInsets = Platform.OS === 'ios' && deviceInfoModule.hasNotch();

export const scaleSize = (size: number): number => (DeviceWidth / guidelineBaseWidth) * size;

export default {
  sizes,
  lineHeights,
  deviceDimensions,
  sizeButton,
  letterSpacing,
  headerHeight,
  homeHeaderHeight,
  primaryItemHeight,
  avatarSizes,
  avatarBorderWidth,
  commentBarHeight,
  postToolbarHeight,
  maxNewsfeedWidth,
  bottomBarHeight,
  deviceWidth: DeviceWidth,
  deviceHeight: DeviceHeight,
  isPhoneWithInsets,
};
