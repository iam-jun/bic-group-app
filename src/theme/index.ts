export { default as fonts } from './fonts';
export { default as spacing } from './spacing';
export { default as dimension } from './dimension';
export { default as shadow } from './shadow';
