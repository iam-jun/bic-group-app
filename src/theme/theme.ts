import { DefaultTheme, DarkTheme, ExtendedTheme } from '@react-navigation/native';
import elevations from './elevations';

export const light: ExtendedTheme = {
  dark: false,
  colors: {
    ...DefaultTheme.colors,
    black: '#000000',
    white: '#FFFFFF',
    danger: '#F64E60',
    success: '#13B055',
    warning: '#FFBC1F',

    neutral: '#FFFFFF',
    neutral1: '#F8F9FF',
    neutral2: '#F1F2F8',
    neutral5: '#E3E6F2',
    neutral10: '#D6D9EB',
    neutral20: '#BAC0DE',
    neutral30: '#9FA6D1',
    neutral40: '#353E6E',
    neutral50: '#2E3660',
    neutral60: '#282E53',
    neutral70: '#212745',
    neutral80: '#181C32',
    neutral90: '#0D0F1C',

    gray1: '#F2F5F8',
    gray5: '#EAEDF2',
    gray10: '#DDDFE4',
    gray20: '#D1D4DB',
    gray30: '#BABEC9',
    gray40: '#A4A9B7',
    gray50: '#767D93',
    gray60: '#515667',
    gray70: '#3F4350',
    gray80: '#2D3039',
    gray90: '#1B1D22',

    purple1: '#FBF8FF',
    purple2: '#F4EFFB',
    purple5: '#E9DFF6',
    purple10: '#C9AFE9',
    purple20: '#B48FE0',
    purple30: '#9E6FD8',
    purple40: '#884FCF',
    purple50: '#7335C0',
    purple60: '#602CA0',

    blue1: '#F5FAFF',
    blue2: '#EBF4FF',
    blue5: '#D6EAFF',
    blue10: '#ADD5FF',
    blue20: '#85C0FF',
    blue30: '#5CABFF',
    blue40: '#3396FF',
    blue50: '#007AFF',
    blue60: '#006CE0',

    red1: '#FEF3F4',
    red2: '#FEECED',
    red5: '#FDD8DC',
    red10: '#FCC5CA',
    red20: '#F1999F',
    red30: '#F0878D',
    red40: '#F64E60',
    red50: '#F42A3E',
    red60: '#E90C22',

    green1: '#FAFFFC',
    green2: '#EFFDF4',
    green5: '#E5F9EC',
    green10: '#D0F1DB',
    green20: '#8ED6A4',
    green30: '#74C18C',
    green40: '#56B374',
    green50: '#13B055',
    green60: '#0E9B49',

    slate1: '#EEEDFD',
    slate5: '#DDDAFB',
    slate10: '#CCC8F9',
    slate20: '#AAA3F5',
    slate30: '#8A81F7',
    slate40: '#7B71F0',
    slate50: '#6559EE',
    slate60: '#4334EA',

    violet1: '#F1F1F9',
    violet5: '#E3E4F3',
    violet10: '#D4D6EC',
    violet20: '#B8BBE0',
    violet30: '#A1A6DC',
    violet40: '#959AD1',
    violet50: '#7F85C7',
    violet60: '#636ABB',

    turquoise1: '#EDFDFC',
    turquoise5: '#DBFAF9',
    turquoise10: '#B7F5F2',
    turquoise20: '#93F0EC',
    turquoise30: '#5DE9E2',
    turquoise40: '#28E2D8',
    turquoise50: '#1BC5BD',
    turquoise60: '#18B4AC',

    transparent: 'rgba(255,255,255,0)',
    transparent1: 'rgba(17, 21, 26, 0.6)',
  },
  elevations: elevations.light,
};

export const dark: ExtendedTheme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    black: '#000000',
    white: '#FFFFFF',
    danger: '#F64E60',
    success: '#13B055',
    warning: '#FFBC1F',

    neutral: '#FFFFFF',
    neutral1: '#F8F9FF',
    neutral2: '#F1F2F8',
    neutral5: '#E3E6F2',
    neutral10: '#D6D9EB',
    neutral20: '#BAC0DE',
    neutral30: '#9FA6D1',
    neutral40: '#353E6E',
    neutral50: '#2E3660',
    neutral60: '#282E53',
    neutral70: '#212745',
    neutral80: '#181C32',
    neutral90: '#0D0F1C',

    gray1: '#F2F5F8',
    gray5: '#EAEDF2',
    gray10: '#DDDFE4',
    gray20: '#D1D4DB',
    gray30: '#BABEC9',
    gray40: '#A4A9B7',
    gray50: '#767D93',
    gray60: '#515667',
    gray70: '#3F4350',
    gray80: '#2D3039',
    gray90: '#1B1D22',

    purple1: '#FBF8FF',
    purple2: '#F4EFFB',
    purple5: '#E9DFF6',
    purple10: '#C9AFE9',
    purple20: '#B48FE0',
    purple30: '#9E6FD8',
    purple40: '#884FCF',
    purple50: '#7335C0',
    purple60: '#602CA0',

    blue1: '#F5FAFF',
    blue2: '#EBF4FF',
    blue5: '#D6EAFF',
    blue10: '#ADD5FF',
    blue20: '#85C0FF',
    blue30: '#5CABFF',
    blue40: '#3396FF',
    blue50: '#007AFF',
    blue60: '#006CE0',

    red1: '#FEF3F4',
    red2: '#FEECED',
    red5: '#FDD8DC',
    red10: '#FCC5CA',
    red20: '#F1999F',
    red30: '#F0878D',
    red40: '#F64E60',
    red50: '#F42A3E',
    red60: '#E90C22',

    green1: '#FAFFFC',
    green2: '#EFFDF4',
    green5: '#E5F9EC',
    green10: '#D0F1DB',
    green20: '#8ED6A4',
    green30: '#74C18C',
    green40: '#56B374',
    green50: '#13B055',
    green60: '#0E9B49',

    slate1: '#EEEDFD',
    slate5: '#DDDAFB',
    slate10: '#CCC8F9',
    slate20: '#AAA3F5',
    slate30: '#8A81F7',
    slate40: '#7B71F0',
    slate50: '#6559EE',
    slate60: '#4334EA',

    violet1: '#F1F1F9',
    violet5: '#E3E4F3',
    violet10: '#D4D6EC',
    violet20: '#B8BBE0',
    violet30: '#A1A6DC',
    violet40: '#959AD1',
    violet50: '#7F85C7',
    violet60: '#636ABB',

    turquoise1: '#EDFDFC',
    turquoise5: '#DBFAF9',
    turquoise10: '#B7F5F2',
    turquoise20: '#93F0EC',
    turquoise30: '#5DE9E2',
    turquoise40: '#28E2D8',
    turquoise50: '#1BC5BD',
    turquoise60: '#18B4AC',

    transparent: 'rgba(255,255,255,0)',
    transparent1: 'rgba(17, 21, 26, 0.6)',
  },
  elevations: elevations.dark,
};

export default {
  light,
  dark,
};
