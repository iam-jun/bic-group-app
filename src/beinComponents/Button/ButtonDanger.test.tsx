import React from 'react';

import {renderWithRedux, cleanup, fireEvent} from '~/test/testUtils';
import {colors} from '~/theme';

import ButtonDanger from './ButtonDanger';

afterEach(cleanup);

describe('ButtonDanger conponent', () => {
  const children = 'Button Danger';

  const testID = 'button_danger.test';

  it('renders correctly', () => {
    const rendered = renderWithRedux(
      <ButtonDanger testID={testID}>{children}</ButtonDanger>,
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const {getByTestId} = renderWithRedux(
      <ButtonDanger testID={testID}>{children}</ButtonDanger>,
    );
    const component = getByTestId(testID);

    expect(component.props.children[0].props.children[2].props.children).toBe(
      children,
    );
  });

  it('renders style correctly', () => {
    const {getByTestId} = renderWithRedux(
      <ButtonDanger style={{margin: 12}} testID={testID}>
        {children}
      </ButtonDanger>,
    );

    const component = getByTestId(testID);

    expect(component.props.style.margin).toBe(12);
  });

  it('renders disabled correctly', () => {
    const onPress = jest.fn();
    const {getByTestId} = renderWithRedux(
      <ButtonDanger disabled onPress={onPress} testID={testID}>
        {children}
      </ButtonDanger>,
    );

    const component = getByTestId(testID);
    fireEvent.press(component);
    expect(onPress).not.toBeCalled();

    expect(component.props.style.backgroundColor).toBe(
      colors.light.colors.bgDisable,
    );

    expect(component.props.children[0].props.children[2].props.color).toBe(
      colors.light.colors.textDisabled,
    );
  });
});
