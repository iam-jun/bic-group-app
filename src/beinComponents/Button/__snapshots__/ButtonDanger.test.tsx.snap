// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDanger conponent renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 4,
      "padding": 8,
    }
  }
  testID="button_danger.test"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    testID="button_wrapper.content"
  >
    <Text
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {},
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Button Danger
    </Text>
  </View>
</View>
`;
